File: php/transcribe.php
<?php
// Transcribe a video with the VidScribe API.
//
// Usage:  php transcribe.php "https://youtu.be/VIDEO_ID"

$apiKey = getenv("VIDSCRIBE_API_KEY") ?: "tsk_your_secret_key";
$base   = getenv("VIDSCRIBE_BASE_URL") ?: "https://getvidscribe.com/v1";

function vidscribe_request($method, $url, $apiKey, $body = null) {
    $ch = curl_init($url);
    $headers = ["x-api-key: $apiKey"];
    if ($body !== null) {
        $headers[] = "content-type: application/json";
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
    }
    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $resp = curl_exec($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    return [$code, json_decode($resp, true)];
}

function transcribe($url, $apiKey, $base, $lang = "en", $mode = "auto") {
    [$code, $data] = vidscribe_request("POST", "$base/transcript", $apiKey, [
        "url" => $url, "lang" => $lang, "mode" => $mode,
    ]);
    if ($code >= 400) {
        throw new Exception("HTTP $code: " . json_encode($data));
    }
    // Short videos return the transcript; long ones return a job to poll.
    if (!empty($data["transcript"])) {
        return $data["transcript"];
    }
    return poll($data["job_id"], $apiKey, $base);
}

function poll($jobId, $apiKey, $base) {
    while (true) {
        [$code, $data] = vidscribe_request("GET", "$base/transcript/$jobId", $apiKey);
        if ($data["status"] === "completed") {
            return $data["transcript"];
        }
        if ($data["status"] === "failed") {
            throw new Exception($data["error"] ?? "Transcription failed.");
        }
        echo "  job $jobId: {$data['status']} ...\n";
        sleep(3);
    }
}

$videoUrl = $argv[1] ?? "https://youtu.be/VIDEO_ID";
try {
    echo transcribe($videoUrl, $apiKey, $base) . "\n";
} catch (Exception $e) {
    fwrite(STDERR, $e->getMessage() . "\n");
    exit(1);
}

File: ruby/transcribe.rb
# Transcribe a video with the VidScribe API.
#
# Usage:  ruby transcribe.rb "https://youtu.be/VIDEO_ID"

require "net/http"
require "json"
require "uri"

API_KEY = ENV["VIDSCRIBE_API_KEY"] || "tsk_your_secret_key"
BASE = ENV["VIDSCRIBE_BASE_URL"] || "https://getvidscribe.com/v1"

def request(method, url, body = nil)
  uri = URI(url)
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = uri.scheme == "https"
  req = method == :post ? Net::HTTP::Post.new(uri) : Net::HTTP::Get.new(uri)
  req["x-api-key"] = API_KEY
  if body
    req["content-type"] = "application/json"
    req.body = JSON.dump(body)
  end
  res = http.request(req)
  raise "HTTP #{res.code}: #{res.body}" if res.code.to_i >= 400
  JSON.parse(res.body)
end

def transcribe(url, lang: "en", mode: "auto")
  data = request(:post, "#{BASE}/transcript", { url: url, lang: lang, mode: mode })
  # Short videos return the transcript; long ones return a job to poll.
  return data["transcript"] if data["transcript"] && !data["transcript"].empty?
  poll(data["job_id"])
end

def poll(job_id)
  loop do
    data = request(:get, "#{BASE}/transcript/#{job_id}")
    return data["transcript"] if data["status"] == "completed"
    raise(data["error"] || "Transcription failed.") if data["status"] == "failed"
    puts "  job #{job_id}: #{data['status']} ..."
    sleep 3
  end
end

url = ARGV[0] || "https://youtu.be/VIDEO_ID"
begin
  puts transcribe(url)
rescue => e
  warn e.message
  exit 1
end
